feat: Release notes for Codacy Self-hosted 13.0.0 (#1913)
codacy/docs: docs/release-notes/self-hosted/self-hosted-v13.0.0.md

---

# Self-hosted v13.0.0

These release notes are for [Codacy Self-hosted v13.0.0](https://github.com/codacy/chart/releases/tag/13.0.0){: target="_blank"}, released on October 26, 2023.<!-- TODO Update release date -->

📢 [Visit the Codacy roadmap](https://roadmap.codacy.com) and <span class="skip-vale">let us know</span> your feedback on both new and planned product updates!

## Upgrading Codacy Self-hosted

Follow the steps below to upgrade to Codacy Self-hosted v13.0.0:

1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.

1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v13.0/chart/maintenance/upgrade/).

1.  Update your Codacy command-line tools to the following versions:

    -   [Codacy Analysis CLI 7.9.4](https://github.com/codacy/codacy-analysis-cli/releases/tag/7.9.4)
    -   [Codacy Coverage Reporter 13.13.8](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.13.8)

> ## Support for ingress-nginx 4.8.3

Reviewer: @tfrgoncalves @loliveiracodacy Can you please validate this **Support for ingress-nginx 4.8.3** section? 🙏

Reviewer 2: For correctness sake, 4.8.3 is the chart version which installs the Nginx version (1.9.4) fixing the issues.
Overall we should refer to it as "Support for ingress-nginx chart 4.8.3". I wouldn't propose mentioning the Nginx fix version (1.9.4) for simplicity,

Reviewer: ```suggestion
## Support for ingress-nginx chart 4.8.3
```

This version of Codacy Self-hosted adds support for [ingress-nginx 4.8.3](https://github.com/kubernetes/ingress-nginx/releases/tag/helm-chart-4.8.3), which fixes the following vulnerabilities:

-   [CVE-2022\-4886](https://github.com/kubernetes/ingress-nginx/issues/10570)
-   [CVE-2023\-5043](https://github.com/kubernetes/ingress-nginx/issues/10571)
-   [CVE-2023\-5044](https://github.com/kubernetes/ingress-nginx/issues/10572)

!!! note
    Although [ingress-nginx 4.8.3 officially supports](https://github.com/kubernetes/ingress-nginx/tree/helm-chart-4.8.3#supported-versions-table) Kubernetes **version 1.25** to **1.28**, we tested this version of Codacy Self-hosted using [ingress-nginx 4.8.3](https://github.com/kubernetes/ingress-nginx/releases/tag/helm-chart-4.8.3) on all the [supported Kubernetes versions](https://docs.codacy.com/v13.0/chart/requirements/#kubernetes-or-microk8s-cluster-setup), **1.22.\*** to **1.26.\***. All tests were successful in accordance with our quality assurance standards.

### Upgrading ingress-nginx to version 4.8.3

!!! warning

>     -   If you applied to ingress-nginx **custom configurations** different from the Codacy's `values-nginx.yaml` file, follow the [ingress-nginx documentation on how to upgrade your installation](https://github.com/kubernetes/ingress-nginx/blob/main/docs/deploy/upgrade.md).

Reviewer: @tfrgoncalves Is this the correct ingress-nginx upgrade link to point to?

Reviewer 2: I haven't found any specific helpful link for upgrading. We could instead point to the chart release:
... follow [ingress-nginx documentation](https://artifacthub.io/packages/helm/ingress-nginx/ingress-nginx/4.8.3)

Reviewer: ```suggestion
    -   If you applied **custom configurations** to ingress-nginx different from the Codacy's `values-nginx.yaml` file, follow the [ingress-nginx documentation](https://artifacthub.io/packages/helm/ingress-nginx/ingress-nginx/4.8.3) to upgrade your installation.
```

    -   If your NGINX Ingress Controller is shared by other components out of Codacy's installation scope, check those components documentation to evaluate the impact of this upgrade.

    If you have any questions regarding the above scenarios or need help, please contact <mailto:support@codacy.com>.

If your Codacy Self-hosted setup uses a **dedicated NGINX Ingress Controller** that follows [Codacy's installation instructions](https://docs.codacy.com/v13.0/chart/infrastructure/eks-quickstart/) with **no custom configurations**, you can upgrade ingress-nginx to version 4.8.3 by executing the steps below:

1.  Download the updated configuration file [`values-nginx.yaml`](https://docs.codacy.com/v13.0/chart/values-files/values-nginx.yaml) for the NGINX Ingress Controller, which enables `enableAnnotationValidations`.

1.  Run the following script:

    ```
    kubectl create namespace codacy
    helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
    helm repo update
    helm upgrade --install --namespace codacy --version 4.8.3 nginx-ingress ingress-nginx/ingress-nginx -f values-nginx.yaml
    ```

> This upgrade can also be applied to supported Codacy Self-Hosted versions before v13.0.

Reviewer: @tfrgoncalves I'm not sure if we should keep this sentence... 🤔 

Have we tested the previous Codacy Self-Hosted versions to ensure they all work correctly? (for example, [here we say](https://github.com/codacy/docs/pull/1913/files#diff-0f076c5cec766edb397917269450edec8fe97ff047abc2b4d5bdba867b633128R37) that we tested all the supported Kubernetes versions)

Thoughts?
cc @loliveiracodacy @Ngwerume

Reviewer 2: We have tested from SH 10 to the most recent one and all is working accordingly. Instead of “This upgrade can also be applied to supported Codacy Self-Hosted versions before v13.0.” we should go with something like “This upgrade can also be applied to supported Codacy Self-Hosted v10.0 and after.”

Reviewer: ```suggestion
!!! note
    This ingress-nginx chart upgrade can be applied to Codacy Self-hosted v10.0 and later.
```

## Bug fixes

-   Pylint now supports Python 3.11 syntax. (TS-467)
-   Added support for the ESLint plugin [<span class="skip-vale">vue-scoped-css</span>](https://www.npmjs.com/package/eslint-plugin-vue-scoped-css). (TS-356)
-   Added support for the ESLint plugin [<span class="skip-vale">eslint-plugin-rxjs</span>](https://www.npmjs.com/package/eslint-plugin-rxjs). (TS-346)
-   Added support for the ESLint plugins [<span class="skip-vale">eslint-plugin-markdown</span>](https://www.npmjs.com/package/eslint-plugin-markdown) and [<span class="skip-vale">eslint-plugin-yml</span>](https://www.npmjs.com/package/eslint-plugin-yml). (TS-121)
-   Fixed an incorrect popup being shown when loading the repository Code patterns page. (PLUTO-717)

## Tool versions

This version of Codacy Self-hosted includes the tool versions below. The tools that were updated on this version are highlighted in bold:

-   Ameba 1.4.3
-   **[Bandit 1.7.5](https://github.com/PyCQA/bandit/releases/tag/1.7.5) (updated from 1.7.0)**
-   Brakeman 4.3.1

> -   **bundler-audit (deprecated) 0.9.1 (updated from 0.6.1)**

Reviewer: @stefanvacareanu7 What's the correct way to present the deprecated tools? Using the old or the new version? 🤔 

Using the old version:
```suggestion
-   **bundler-audit (deprecated) 0.6.1**
```

Reviewer: Or using the new version:
```suggestion
-   **bundler-audit (deprecated) 0.9.1**
```

Reviewer: @nicklem clarified to me that we updated some of the tools we are deprecating, so I updated it accordingly (keeping the new and old versions).
